dartagnan/src/main/java/com/dat3m/dartagnan/utils/visualization/ExecutionGraphVisualizer.java

```diff
@@ -89,10 +87,9 @@ public void generateGraphOfExecutionModel(Writer writer, String graphName, Execu
     }
 
     private boolean ignore(EventData e) {
-        return !e.getEvent().hasMetadata(SourceLocation.class) && !e.isInit();
+        return false; // We ignore no events for now.
```

Comment:
isn't this affection our current witness generation?

Reply:
Some addresses get a `null` representation in graphviz witnesses, see e.g. ticketlock with `-DACQ2RX`

Reply:
Hmm, the `NULL` entries should already have been there before.
It seems we simply do not construct init events for the members of the `lock` variable, cause Smack fails to init properly.
That problem should already exist on `development`. 

While debugging, I noticed that we now do create a lot of memory objects related to smack internals.
I will try to fix this.

Reply:
I fixed both issues. We now assume that Smack generated correct static initialization code IFF it generated at least one init value. This might be wrong if there is the possibility that Smack generates some but not all initialization of a memory object (i.e., it generates partial initialization).